src/polyswarmd/websockets/filter.py

```diff
@@ -1,40 +1,44 @@
+import gevent
 import logging
 import weakref
-from collections import namedtuple
 from contextlib import contextmanager
-from typing import Any, Callable, Collection, Container, Type
-from random import gauss
-
-from requests.exceptions import ConnectionError
-
-import gevent
-import web3.eth
 from gevent.pool import Group
 from gevent.queue import Queue
+from random import gauss
+from requests.exceptions import ConnectionError
+from typing import Collection, Type
 from web3.utils.filters import LogFilter
 
+
 from . import messages
 
 logger = logging.getLogger(__name__)
 
 
-class FilterWrapper(namedtuple('Filter', ['filter', 'formatter', 'backoff'])):
-    "A utility class which wraps a contract filter with websocket-messaging features"
+class FilterWrapper:
+    """A utility class which wraps a contract filter with websocket-messaging features"""
+    # Allow this class to be weak referencable
+    __slots__ = ('__weakref__', 'filter', 'formatter', 'backoff')
```

Comment:
We no longer _rely_ on `weakref` because we serve items through a `contextmanager`. I think it might make more sense for us to just feed in `self.wrappers` directly and drop use of `weakref` entirely.

Reply:
Alright, I will test that out